minitrace.c

```diff
@@ -244,11 +248,21 @@ void mtr_flush() {
 	// We have to lock while flushing. So we really should avoid flushing as much as possible.
 
 	pthread_mutex_lock(&mutex);
-	int old_tracing = is_tracing;
-	is_tracing = 0;	// Stop logging even if using interlocked increments instead of the mutex. Can cause data loss.
+	// use double_buffering for flushing: we create two buffers and swap them here on every flush
+	// thus flushing and tracing can happen concurrently without them intermingeling
+	// note: this assumes that only one flush can be active at any point in time!
```

Comment:
how is this ensured?

Reply:
Currently there is nothing ensuring only one flush happens at a time.
We could add a second mutex for this that wraps the whole flush function.

Reply:
Yeah, sounds reasonable. To test that it works just make the buffer size really small and add a Sleep(1000) in flush :)

Once you've got this working, I'm happy to merge this.